store/store.go

```diff
@@ -123,6 +123,17 @@ func (a *Alerts) Delete(fp model.Fingerprint) error {
 	return nil
 }
 
```

Comment:
If we chose to go with my approach or your approach - we'd need to remove the `Delete` method as it is no longer used.

Reply:
I think the same holds true for `Get`, which is also used in `store` but I don't see it used anywhere else (but tests).

Reply:
I don't have a strong opinion here, and I'm happy to remove `Get` and `Delete`, but I did just want to raise a counter-argument for this to see if we still think it makes sense.

I understand that `Get` and `Delete` will be un-used once we merge this PR, but does that mean we should delete them? These operations seem kind of important for a store interface, even if we don't make use of them today. It's more of a question of semantics rather than pragmatics, but I feel it's important to at least discuss it.

Reply:
OK! 🙂 `Get` and `Delete` have been deleted! 👍

Reply:
Oops! It turns out I cannot remove `Get` because it's used in `provider/mem/mem.go`.

Reply:
Yeah, but if you look at the places where that Get is used you'll realise it's only in tests.

Reply:
It's also used on Line 207 of `provider/mem/mem.go` which isn't a test?

Reply:
Sorry, I could have been clearer. I meant the `Get` on this file. 

```diff
diff --git a/provider/mem/mem.go b/provider/mem/mem.go
index cfc3bfc3..b84133e4 100644
--- a/provider/mem/mem.go
+++ b/provider/mem/mem.go
@@ -20,12 +20,10 @@ import (
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/common/model"
-
 	"github.com/prometheus/alertmanager/provider"
 	"github.com/prometheus/alertmanager/store"
 	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 const alertChannelLength = 200
@@ -190,11 +188,6 @@ func (a *Alerts) GetPending() provider.AlertIterator {
 	return provider.NewAlertIterator(ch, done, nil)
 }
 
-// Get returns the alert for a given fingerprint.
-func (a *Alerts) Get(fp model.Fingerprint) (*types.Alert, error) {
-	return a.alerts.Get(fp)
-}
-
 // Put adds the given alert to the set.
 func (a *Alerts) Put(alerts ...*types.Alert) error {
 	for _, alert := range alerts {

```

Reply:
Discussed this offline, we can't remove it because due to the interface this struct implements.